File: exercismdotio/python/bob/bob.py
#
# Skeleton file for the Python "Bob" exercise.
#


def hey(what):
	what = what.strip()
	if what == '':
		return 'Fine. Be that way!'		
	elif what == what.upper() and any([character.isalpha() for character in what]):
		return 'Whoa, chill out!'
	elif what.endswith('?'):
		return 'Sure.'
	else:
		return 'Whatever.'

File: exercismdotio/javascript/bob/bob.js
//
// This is only a SKELETON file for the "Bob" exercise. It's been provided as a
// convenience to get you started writing code faster.
//

var Bob = function() {};

Bob.prototype.hey = function(input) {
    input = input.trim();

    if (input === '') {
        return 'Fine. Be that way!';
    }

    if (input === input.toUpperCase() && input.match(/[A-Za-z\u00C0-\u017F]/)) {
        return 'Whoa, chill out!';
    }

    if (input[input.length - 1] === '?') {
        return 'Sure.';
    }

    return 'Whatever.';
};

module.exports = Bob;
